perfkitbenchmarker/providers/gcp/gcp_dpb_dataflow_template.py

```diff
@@ -0,0 +1,153 @@
+# limitations under the License.
+"""Module containing class for GCP's Dataflow service.
+Use this module for running Dataflow jobs from pre-built Dataflow templates
+such as https://cloud.google.com/dataflow/docs/guides/templates/provided-templates
+
+No Clusters can be created or destroyed, since it is a managed solution
+See details at: https://cloud.google.com/dataflow/
+"""
+
+import os
+import re
+import time
+import json
+import logging
+import datetime
+
+from absl import flags
+from perfkitbenchmarker import beam_benchmark_helper
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import providers
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.providers.gcp import gcp_dpb_dataflow
+from perfkitbenchmarker.providers.gcp import util
+
+# Refer to flags with prefix 'dpb_df_template' defined in gcp provider flags.py
+FLAGS = flags.FLAGS
+
+class GcpDpbDataflowTemplate(gcp_dpb_dataflow.GcpDpbDataflow):
+  """Object representing GCP Dataflow service for running job templates."""
+
+  CLOUD = providers.GCP
+  SERVICE_TYPE = 'dataflow_template'
+
+  def __init__(self, dpb_service_spec):
+    super(GcpDpbDataflowTemplate, self).__init__(dpb_service_spec)
+    self.project = util.GetDefaultProject()
+    self.input_sub_empty = False
+    self.job_drained = False
+
+  @staticmethod
+  def CheckPrerequisites(benchmark_config):
+    del benchmark_config  # Unused
+    if not FLAGS.dpb_df_template_gcs_location:
+      raise errors.Config.InvalidValue('Template GCS location missing.')
+
+  def Create(self):
+    """See base class."""
+    pass
+
+  def Delete(self):
+    """See base class."""
+    pass
+
+  def SubmitJob(
+      self,
+      template_gcs_location=None,
+      job_poll_interval=None,
+      job_arguments=None,
+      job_input_sub = None):
+
+    worker_machine_type = self.spec.worker_group.vm_spec.machine_type
+    num_workers = self.spec.worker_count
+    max_workers = self.spec.worker_count
+
+    now = datetime.datetime.now()
+    job_name = template_gcs_location.split('/')[-1] \
+            + '_' + now.strftime("%Y%m%d_%H%M%S")
+    region = util.GetRegionFromZone(FLAGS.dpb_service_zone)
+
+    cmd = util.GcloudCommand(self, 'dataflow', 'jobs', 'run', job_name)
+    cmd.flags = {
+        'project': self.project,
+        'gcs-location': template_gcs_location,
+        'staging-location': FLAGS.dpb_dataflow_temp_location,
+        'region': region,
+        'worker-region': region,
+        'worker-machine-type': worker_machine_type,
+        'num-workers': num_workers,
+        'max-workers': max_workers,
+        'parameters': ','.join(job_arguments),
+        'format': 'json',
+    }
+
+    stdout, _, _ = cmd.Issue()
+
+    # Parse output to retrieve submitted job ID
+    try:
+      result = json.loads(stdout)
+      self.job_id = result['id']
+    except Exception as err:
+      logging.error('Failed to parse Dataflow job ID: {}'.format(err))
```

Comment:
Don't interpolate variables directly in strings to be logged. Refer to https://google.github.io/styleguide/pyguide.html#3101-logging